src/utils/types.rs

```diff
@@ -1,10 +1,45 @@
-pub(crate) const DATA_LENGTH: usize = 48;
-
-/// Type of dataset's elements to lower bound
-pub(crate) type Element = [u8; DATA_LENGTH];
+//! Types and implementation
+use std::mem;
 
 /// Digest size for internal hashes
 pub(crate) const DIGEST_SIZE: usize = 32;
 
 /// Hash type for internal hashes
 pub(crate) type Hash = [u8; DIGEST_SIZE];
+
+/// Trait to represent input as bytes
+pub trait ToBytes {
```

Comment:
What is the advantage of defining this trait compared to requiring the element type to implement `AsRef<[u8]>` directly?

Reply:
I was not aware of this trait (I found this ToBytes method in several posts online)! I will use this instead! :)